crates/wash-cli/src/app/mod.rs

```diff
@@ -412,37 +451,94 @@ async fn delete_application_version(cmd: DeleteCommand) -> Result<CommandOutput>
-        .await
-        .is_ok_and(|exists| exists)
-    {
-        let manifest = load_app_manifest(cmd.app_name.parse()?)
-            .await
-            .with_context(|| format!("failed to load app manifest at [{}]", cmd.app_name))?;
-        (
-            manifest
-                .name()
-                .map(ToString::to_string)
-                .context("failed to find name of manifest")?,
-            manifest.version().map(ToString::to_string),
-        )
-    } else {
-        (cmd.app_name, cmd.version)
+    // Determine which models to remove, if a single model is not specified,
+    // then attempt to filter the list of existing models
+    let models = match cmd.app_name {
+        // If an explicit app name was specified, resolve the right app name and version
+        Some(app_name) => {
+            // If we have received a valid path to a model file, then read and extract the model name,
+            // otherwise use the supplied name as a model name
+            let (model_name, version): (String, Option<String>) =
+                if tokio::fs::try_exists(&app_name)
+                    .await
+                    .is_ok_and(|exists| exists)
+                {
+                    let manifest = load_app_manifest(app_name.parse()?)
+                        .await
+                        .with_context(|| format!("failed to load app manifest at [{app_name}]"))?;
+                    (
+                        manifest
+                            .name()
+                            .map(ToString::to_string)
+                            .context("failed to find name of manifest")?,
+                        manifest.version().map(ToString::to_string),
+                    )
+                } else {
+                    (app_name, cmd.version)
+                };
+
+            vec![(model_name, version)]
+        }
+        // If no model name was specified, use command-specified filters to determine which models to act on
+        None if cmd.all_undeployed => wash_lib::app::get_models(&client, lattice.clone())
+            .await?
+            .into_iter()
+            .filter_map(|m| match m.detailed_status.info.status_type {
+                wadm_types::api::StatusType::Undeployed => Some((m.name, Some(m.version))),
+                _ => None,
+            })
+            .collect(),
+        _ => Vec::new(),
     };
 
-    let deleted =
-        wash_lib::app::delete_model_version(&client, lattice, &model_name, version).await?;
+    let mut deleted_models = Vec::new();
 
-    let mut map = HashMap::new();
-    map.insert("deleted".to_string(), json!(deleted));
-    let message = if deleted {
-        format!("Deleted application: {model_name}")
-    } else {
-        format!("Already deleted application: {model_name}")
+    #[derive(serde::Serialize)]
+    struct ModelNameAndVersion<'a> {
+        model_name: &'a String,
+        version: &'a Option<String>,
+    }
```

Comment:
Any reason why we shouldn't omit this in favor of just a tuple? I assume clarity in the serialized output, but wanted to make sure it was needed

Reply:
Ah so I wanted to make sure that we put out an *object* -- in the future if we expand the information we return here, it will almost surely be in object form -- a tuple is serialized as a list, so this ends up making a list of lists. 

The change of shape would be annoying to change in downstream things that read this output, but if we do it right we can just *add* fields and not have to change the shape of the output